bot.go

```diff
@@ -507,3 +511,45 @@ func (b *Bot) Respond(query Query, results []Result) error {
 
 	return nil
 }
+
+func (b *Bot) Handle(command string, handler Handler) {
+	if command == Default {
+		b.DefaultHandler = handler
+		return
+	}
+
+	b.Handlers[command] = handler
+}
+
+func (b *Bot) Serve() {
+	messages := make(chan Message)
+	b.Listen(messages, 1*time.Second)
+
+	for message := range messages {
+		if handler, args, err := b.route(&message); err == nil {
+			handler(Context{Message: &message, Args: args})
+		}
+	}
+}
+
+func (b *Bot) route(message *Message) (Handler, map[string]string, error) {
+	for i, handler := range b.Handlers {
+		r, err := regexp.Compile(i)
+		if err != nil {
+			return nil, nil, err
```

Comment:
Err is being returned and not getting handled properly at :529. If I were you I would have compiled all of the regular expressions in `Serve()` inb4 actually serving and used `MustCompile()` in order to get a straight panic (that's what you typically do with regexps.